index.js

```diff
@@ -1,21 +1,30 @@
-import isUndefined from 'lodash/lang/isUndefined';
+'use strict';
+
+var isUndefined = require('lodash.isundefined');
 
 /**
  * Recursive dispatch, returns a function which iterates a series of commands
  * looking for one to handle the target and return a value. The commands adhere
  * to an interface of (target, fn) where the returned dispatch fn is passed
  * along to each command, where it can be used by the command. If the target
- * cannot be handled by a command the command returns undefined.
+ * cannot be handled by a command the command returns undefined. If none of the
+ * supplied commands handle the target, an error is thrown.
  *
  * @param commands
  * @returns {Function}
  */
-function dispatch(...commands) {
+
+module.exports = function dispatch() {
+  var commands = new Array(arguments.length);
```

Comment:
Did you have a look at that Bluebird doc I posted? Apparently that leaks the `arguments` object, which causes the V8 engine to deoptimise the function. Tbh, without any perf tests it's probably a premature optimisation.